### PR TITLE
Virus symptoms' statistics now get a random modifier when generated.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Virus symptoms' statistics now get a random modifier when generated.
add: PanD.E.M.I.C. now tells you the total stats of viruses inside.
/:cl:

Stealth and Resistance can get from -1 to +1, and Stage Speed and Transmittability from -2 to +2. If you want a different stat set, you'll have to find that symptom again.
This could allow some more freedom in creating viruses, moving away from pre-determined virus "recipes".

### CODE REVIEW

**Comment by Giacom** [comment]
Not a big fan of adding more randomness to virus creation as it sounds frustrating. I like the idea of having recipes though with viruses even if you follow a recipe you still have to randomly get the correct symptom which you wanted.

**Comment by chickenboy10** [comment]
Im fine with the pandemic change, but not with the random stat change.
Like Giacom said, theres already a huge amount of randomness with getting the right syptom in the first place, adding more randomness on top will just be an extra pain.

**Comment by XDTM** [comment]
First off, this only affects harmful viruses, as healing viruses need no stats. 
That said, getting the correct symptom is not RNG itself, just a matter of time dictated by RNG - eventually you'll have it, the only question is when. Randomizing symptom stats allows you to choose flexibly what you want to make once you have several symptoms: do i have a really high-resistance Cough? Is it worth using, even if it lowers Stealth by 2?
Therefore adding an element of decision-making that is not just planned with a spreadsheet.

On top of that, it also reduces the frustration from getting the same symptom multiple times, which happens quite a lot considering BYOND's RNG.

File: code/datums/diseases/advance/symptoms/symptoms.dm [hunk 1]
```diff
@@ -21,12 +21,20 @@ var/global/const/SYMPTOM_ACTIVATION_PROB = 5
 
 /datum/symptom/New()
 	var/list/S = list_symptoms
+
+	//Randomize Symptom Stats
+	stealth += rand(-1,1)
```
Comment (Giacom) [inline code/datums/diseases/advance/symptoms/symptoms.dm]: This is going to randomise the stats everytime the disease is transferred between bottles or when infecting new people.

Comment (XDTM) [inline code/datums/diseases/advance/symptoms/symptoms.dm]: I tested, it does not randomize with bottle transfer. I'll check on infection.

Comment (XDTM) [inline code/datums/diseases/advance/symptoms/symptoms.dm]: Tested for infection, stats remain constant on infectees.

Comment (Giacom) [inline code/datums/diseases/advance/symptoms/symptoms.dm]: Not sure how that is possible, the advance diseases' New() proc has a disease parameter which it copies the symptoms from by instantiating new symptom datums, which would change the levels in the New().

Comment (XDTM) [inline code/datums/diseases/advance/symptoms/symptoms.dm]: The AddDisease proc in _MobProcs.dm copies any modifications made to the virus on infection; it was previously used so you could varedit viruses and keep the changes.

**Comment by XDTM** [comment]
Replacing new S.type with S did the trick, but i'm not 100% sure it's a good thing to do.
In testing the viruses behaved as they should by mutating, mixing, infecting and devolving; however, varediting a symptom edited all instances of that symptom _who had the same origin_. 
Which means that moving a symptom with a dropper will cause both viruses to share the same symptom, while generating the same symptom through mutation creates two independent instances of the symptom.

**Comment by Giacom** [comment]
I'm not too sure either. The symptoms will now be pointing to the symptom on the origin disease. I would heavily test that if you an origin disease and a copy then if you change the origin disease it will not affect the copies.

Before the copying the disease would copy the symptoms so each disease were the only one with references to their symptom:

image

But I think the way you're changing it, it'll work like this:

image

You'll have to be careful about this.

**Comment by XDTM** [comment]
I'm trying to set up a symptom Copy proc to avoid this

**Comment by XDTM** [comment]
Finally got the bloody thing to work. Also since i was there i made a list/properties var for diseases instead of calculating them from symptoms each time it's called.

**Comment by Niknakflak** [comment]
I'm all for changing virology because it's not that great or fun to do, but all this does is make it WORSE and even LESS fun to do.

**Comment by XDTM** [comment]
The people have voted.

**Comment by Giacom** [comment]
The Pandemic changes aren't a problem. Maybe you could re-open/open a new PR with just the pandemic changes?

**Pull request closed without merging**